subprojects/core-api/src/main/java/org/gradle/api/artifacts/query/ArtifactResolutionQuery.java

```diff
@@ -80,9 +81,9 @@ public interface ArtifactResolutionQuery {
      * @param componentType The expected type of the component.
      * @param artifactTypes The artifacts to retrieve for the queried components.
      */
+    @SuppressWarnings("unchecked")
```

Comment:
I don't see a warning in IDEA. What is the problem here?

Reply:
`Type safety: Potential heap pollution via varargs parameter elements`. These warnings are not visible in IDEA but appear in compiler output.

Reply:
Ah, I see. I didn't realize you need to use `@SuppressWarnings("unchecked")` on an interface for this kind of warning.